x-pack/plugins/enterprise_search/server/lib/indices/utils/get_index_data.ts

```diff
@@ -84,3 +85,58 @@ export const getIndexDataMapper = (totalIndexData: TotalIndexData) => {
       indexName
     );
 };
+
+function isHidden(index: IndicesIndexState): boolean {
+  return Boolean(index.settings?.index?.hidden) || index.settings?.index?.hidden === 'true';
+}
+
+export const getIndexData = async (
+  client: IScopedClusterClient,
+  onlyShowSearchOptimizedIndices: boolean,
+  returnHiddenIndices: boolean,
+  searchQuery?: string
+) => {
+  const expandWildcards: ExpandWildcard[] = returnHiddenIndices ? ['hidden', 'all'] : ['open'];
+  if (onlyShowSearchOptimizedIndices) {
+    const names = await fetchConnectorIndexNames(client);
+    const allIndexNames = searchQuery
+      ? names.filter((indexName) => indexName.toLowerCase().includes(searchQuery.toLowerCase()))
+      : names;
+    const allIndexMatches = await client.asCurrentUser.indices.get({
+      // only get specified index properties from ES to keep the response under 536MB
+      // node.js string length limit: https://github.com/nodejs/node/issues/33960
+      filter_path: ['*.aliases', '*.settings.index.hidden'],
+      ignore_unavailable: true,
```

Comment:
an issue here where if theres no indices, the indices target will be empty. I was getting a uncaught exception here as ignore_unavailable complains.

This is just enriching the indices metadata? Could just return early if array is empty.

Reply:
Good catch! Fixed :)